src/components/KonnectorFolder.jsx

```diff
@@ -1,25 +1,87 @@
-import React from 'react'
+import React, { Component, PureComponent } from 'react'
+import classNames from 'classnames'
+
+import Icon from 'cozy-ui/react/Icon'
 import { translate } from 'cozy-ui/react/I18n'
 
 import styles from '../styles/konnectorFolder'
 
 import DescriptionContent from './DescriptionContent'
 
-class KonnectorFolder extends React.Component {
-  render({ t, driveUrl, trigger }) {
+const ERRORED = 'ERRORED'
+const FETCHING = 'FETCHING'
+const SUCCESS = 'SUCCESS'
+
+/**
+ * Cache
+ */
+let driveUrl
+
+/**
+ * Renders a link only if href prop is provided
+ * @extends PureComponent
+ */
+class SmartLink extends PureComponent {
+  render() {
+    const { className, href } = this.props
+    return href ? (
+      <a className={className} href={href}>
+        {this.props.children}
+      </a>
+    ) : (
+      <span className={className}>{this.props.children}</span>
+    )
+  }
+}
+
+export class KonnectorFolder extends Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      status: FETCHING
+    }
+  }
+
+  async componentDidMount() {
+    if (!driveUrl) {
+      try {
+        const { client } = this.context
+        const { data } = await client.collection('io.cozy.apps').all()
+        const driveApp = data.find(app => app.slug === 'drive')
+        driveUrl = driveApp && `${driveApp.links.related}#/files/`
+      } catch (error) {
+        this.setState({
+          status: ERRORED
+        })
+      }
+    }
+
+    this.setState({
+      status: SUCCESS
+    })
+  }
+
+  render() {
+    const { t, trigger } = this.props
+    const { status } = this.state
+    const disabled = status !== SUCCESS || !driveUrl
+    const fetching = status === FETCHING
     return (
       <div className={styles['col-account-folder']}>
         <DescriptionContent title={t('account.folder.withoutSettings.title')}>
-          {driveUrl && (
-            <a
-              className={styles['col-account-folder-link']}
-              href={`${driveUrl}${trigger.message.folder_to_save}`}
-            >
-              {t('account.folder.link')}
-            </a>
-          )}
+          <SmartLink
+            className={classNames(styles['col-account-folder-link'], {
+              'u-silver': disabled,
+              'u-c-wait': disabled && fetching,
+              'u-c-not-allowed': disabled && !fetching
+            })}
+            href={driveUrl && `${driveUrl}${trigger.message.folder_to_save}`}
+          >
+            <Icon className="u-mr-half" icon="openwith" />
+            {t('account.folder.link')}
+          </SmartLink>
```

Comment:
I may not have understood something. Here if we do not have `driveUrl`, there will be text like "click here for the folder" and the link won't work ?

Reply:
Link will be in a disabled state.

It will occur only in dev environment, when drive is not installed but you're right, maybe an explicit message should be better.

As for now the link was just displayed or not displayed at all, this is an improvement so I will check what Zoe think about it.

Reply:
![capture d ecran 2019-03-06 a 01 23 03](https://user-images.githubusercontent.com/776764/53884442-e0200400-401b-11e9-8732-0372f596061a.png)